testing-framework/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2023 Polyphene.
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod error;
+
+use cid::{multihash::Code, Cid};
+use fil_builtin_actors_bundle::BUNDLE_CAR;
+use futures::executor::block_on;
+use fvm::{
+    engine::EnginePool,
+    executor::{ApplyKind, DefaultExecutor, Executor},
+    init_actor,
+    machine::{DefaultMachine, Manifest, NetworkConfig},
+    state_tree::{ActorState, StateTree},
+    system_actor,
+};
+use fvm_integration_tests::{
+    dummy::DummyExterns,
+    tester::{Account, IntegrationExecutor},
+};
+use fvm_ipld_blockstore::{Block, Blockstore, MemoryBlockstore};
+use fvm_ipld_car::load_car_unchecked;
+use fvm_ipld_encoding::{serde::Serialize, CborStore};
+use libsecp256k1::{PublicKey, SecretKey};
+use rand::SeedableRng;
+
+use error::{Error, WrapFVMError};
+use fvm_shared::{
+    address::Address, bigint::Zero, econ::TokenAmount, message::Message, state::StateTreeVersion,
+    version::NetworkVersion, ActorID, IPLD_RAW,
+};
+
+// TODO: document purpose.
+const EAM_ACTOR_ID: ActorID = 10;
+const NETWORK_VERSION: NetworkVersion = NetworkVersion::V18;
+const DEFAULT_BASE_FEE: u64 = 100;
+
+/// Main interface to test `Actor`s with Kythera.
+pub struct Tester {
+    // Builtin actors root Cid used in the Machine
+    builtin_actors: BuiltInActors,
+    // Custom code cid deployed by developer
+    code_cids: Vec<Cid>,
+    // State tree constructed before instantiating the Machine
+    state_tree: StateTree<MemoryBlockstore>,
+    // Account used for testing.
+    account: Account,
+    // Id of the main Actor deployed to be tested.
+    main_actor_id: Option<Vec<u8>>,
+}
+
+/// Built-in Actors that are deployed to the testing `StateTree`.
+struct BuiltInActors {
+    root: Cid,
+    manifest: Manifest,
+}
+
+/// WebAssembly Actor.
+pub struct WasmActor {
+    name: String,
+    code: Vec<u8>,
+}
+
+impl WasmActor {
+    /// Create a new WebAssembly Actor.
+    pub fn new(name: String, code: Vec<u8>) -> Self {
+        Self { name, code }
+    }
+}
+
+impl Tester {
+    /// Create a new Kythera Tester.
+    pub fn new() -> Self {
+        let bs = MemoryBlockstore::default();
+
+        // Initialize state tree
+        let mut state_tree = StateTree::new(bs, StateTreeVersion::V5)
+            .expect("Should be able to put the Version in the StateTree");
+
+        let builtin_actors = Self::load_builtin_actors(&mut state_tree);
+
+        let account =
+            Self::create_account(&mut state_tree, *builtin_actors.manifest.get_account_code());
+
+        Self {
+            builtin_actors,
+            code_cids: vec![],
+            state_tree,
+            account,
+            main_actor_id: None,
+        }
+    }
+
+    /// Load the built-in actors into the `Blockstore`.
+    /// And activate them on the `StateTree`.
+    fn load_builtin_actors<B: Blockstore>(state_tree: &mut StateTree<B>) -> BuiltInActors {
```

Comment:
I think this function is incomplete. We are missing quite a few actors from the builtins that need to be instantiated.

We could refer to [Workbench](https://github.com/anorth/fvm-workbench/blob/main/builtin/src/genesis.rs) to help us on that.